### DISCUSSION TITLE
[Feature Request] Localising guest's names

### DISCUSSION THREAD

**Discussion by telk5093** [OpenRCT2/OpenRCT2#15743] [Feature requests]
Currently, we have hard-coded guest's names at https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/localisation/RealNames.cpp
It will be great to localise them.

Of course, it should be able to change the order, such as ``N. Aaron`` since there're languages that last name comes first.

**Comment by LucaRed** [comment]
It's useless from a gameplay perspective, but I like it!

**Comment by groenroos** [comment]
Some discussion around this here: OpenRCT2/objects#347

It was felt that having localised names won't always make sense or be immersive for every scenario - especially those set in a specific country or real-world area.

My suggestion is to have a bank of names for each language, and adding an option for the player to choose whether they'd prefer to see names that always match their chosen UI language, or names that 'fit' the scenario they're playing. This setting should be underneath the language setting, and be visible/available only when we have a bank of names available for the selected language.

We'd have to designate which names are appropriate/'realistic' for each of the RCT1/2/DLC scenarios. However, not all scenarios (esp. RCT1) are unambiguously set in a specific geographic location though, so I wonder what would be the best thing there?

Consequently, the scenario editor should probably also get a setting to let scenario creators choose which set of names are most appropriate for their scenario.

**Reply by sjoerddebruin** [reply]
I think you are correct with the options, one that follows the user preference and one that follows the scenario preference.  The object selection window offers the same set of languages as the options window. If the default language isn't chosen in the scenario and the user has "scenario preference" selected that peep name object will be used.
